lib/modules/datasource/dart-version/index.ts

```diff
@@ -0,0 +1,79 @@
+import { regEx } from '../../../util/regex';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { DartResponse } from './types';
+
+export const stableVersionRegex = regEx(/^\d+\.\d+\.\d+$/);
+
+export class DartVersionDatasource extends Datasource {
+  static readonly id = 'dart-version';
+
+  constructor() {
+    super(DartVersionDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = ['https://storage.googleapis.com'];
+
+  override readonly caching = true;
+
+  private readonly channels = ['stable', 'beta', 'dev'];
+
+  async getReleases({
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+    const result: ReleaseResult = {
+      homepage: 'https://dart.dev/',
+      sourceUrl: 'https://github.com/dart-lang/sdk',
+      registryUrl,
+      releases: [],
+    };
+    try {
+      for (const channel of this.channels) {
+        const resp = (
+          await this.http.getJson<DartResponse>(
+            `${registryUrl}/storage/v1/b/dart-archive/o?delimiter=%2F&prefix=channels%2F${channel}%2Frelease%2F&alt=json`
```

Comment:
This API call is grabbed when visiting [this](https://dart.dev/get-dart/archive) page to download the Dart SDK.

I tried hitting https://storage.googleapis.com/storage/v1/b/dart-archive/o without the query parameters, which returned more detailed results. But the results are paginated, and I'm not sure how to get to the next page.

Reply:
can we change the prefix to only contain the channel name?

Reply:
I've tried and sadly it doesn't work

![image](https://user-images.githubusercontent.com/12210067/206819565-fcddcbaf-ec88-4344-8c00-23395d1701e8.png)

![image](https://user-images.githubusercontent.com/12210067/206819614-daf64def-286b-4939-b6d4-801c9c9c216b.png)